ampersand-search-select.js

```diff
@@ -0,0 +1,241 @@
+        selectState._data.configure({
+          filter: function(model) {
+            var name = model.name.toLowerCase();
+            return transformedQuery.length > 0 ? _.startsWith(name, transformedQuery) || _.startsWith(transformedQuery, name) : false;
+          }
+        }, true);
+
+        selectState._querySet = selectState._data.models;
+      });
+    },
+    selectItem: function(event) {
+      d3.select(event.target.parentNode).each(function(d) {
+        this._selected.push(d[this.idAttribute]);
+        this.trigger('change:_selected', this, this._selected);
+      }.bind(this));
+    },
+    deselectItem: function(event) {
+      d3.select(event.target.parentNode).each(function(d) {
+        _.remove(this._selected, function(id) { return id === d[this.idAttribute]; }.bind(this));
+        this.trigger('change:_selected', this, this._selected);
+      }.bind(this));
+    }
+  });
+
+  var SearchSelectView = AmpersandView.extend({
+    template: '<svg></svg>',
+    autoRender: true,
+    initialize: function() {
+      this.model._view = this;
+    },
+    events: {
+      'keyup .ampersand-search-select-bar': 'search',
+      'click svg.ampersand-search-select-query-add': 'selectItem',
+      'click svg.ampersand-search-select-query-check': 'deselectItem'
+    },
+    bindings: {
+      'model._querySet': {
+        type: function(el, querySet) {
+          this.renderQuerySet(querySet);
+        }
+      },
+      'model._selected': {
+        type: function(el, selected) {
+          this.renderQuerySet(this.model._querySet);
+        }
+      }
+    },
+    render: function() {
+      AmpersandView.prototype.render.call(this);
+
+      var searchSelect = this.svg = d3.select(this.el)
+        .attr('class', 'ampersand-search-select')
+        .attr('width', '100%')
+        .attr('height', '9.5em');
+
+      var searchForeignObject = searchSelect.append('foreignObject')
+        .attr('class', 'ampersand-search-select-foreign-object')
+        .attr('width', '100%');
+
+      var searchBody = searchForeignObject.append('xhtml:body')
+        .attr('class', 'ampersand-search-select-body')
+        .style('position', 'relative');
+
+      searchBody.append('input')
+        .attr('class', 'ampersand-search-select-bar')
+        .attr('type', 'text');
+
+      var searchGlass = searchBody.append('svg')
+        .attr('class', 'ampersand-search-select-glass')
+        .attr('width', 20)
+        .attr('height', 20)
+        .style('position', 'relative');
+
+      searchGlass.append('circle')
+        .attr('class', 'ampersand-search-select-glass-circle')
+        .attr('cx', 10)
+        .attr('cy', 7.5)
+        .attr('r', 5);
+
+      searchGlass.append('line')
+        .attr('class', 'ampersand-search-select-glass-line')
+        .attr('x1', 0)
+        .attr('y1', 17)
+        .attr('x2', 7.5)
+        .attr('y2', 10);
+    },
+    renderQuerySet: function(querySet) {
+      if (this.svg) {
```

Comment:
can you reverse this condition and just return early?  It's hard to follow long spans of indentation like this.